include/alpaka/acc/Tag.hpp

```diff
@@ -0,0 +1,66 @@
+/* Copyright 2022 Simeon Ehrig
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/BoostPredef.hpp>
+
+#include <iostream>
+#include <type_traits>
+
+#define CREATE_ACC_TAG(tag_name)                                                                                      \
+    struct tag_name                                                                                                   \
+    {                                                                                                                 \
+        static std::string get_name()                                                                                 \
+        {                                                                                                             \
+            return #tag_name;                                                                                         \
+        }                                                                                                             \
+    }
+
+namespace alpaka
+{
+    CREATE_ACC_TAG(AccCpuFibersTag);
+    CREATE_ACC_TAG(AccCpuOmp2BlocksTag);
+    CREATE_ACC_TAG(AccCpuOmp2ThreadsTag);
+    CREATE_ACC_TAG(AccCpuSerialTag);
+    CREATE_ACC_TAG(AccCpuSyclIntelTag);
+    CREATE_ACC_TAG(AccCpuTbbBlocksTag);
+    CREATE_ACC_TAG(AccCpuThreadsTag);
+    CREATE_ACC_TAG(AccFpgaSyclIntelTag);
+    CREATE_ACC_TAG(AccFpgaSyclXilinxTag);
+    CREATE_ACC_TAG(AccGenericSyclTag);
+    CREATE_ACC_TAG(AccGpuCudaRtTag);
+    CREATE_ACC_TAG(AccGpuHipRtTag);
+    CREATE_ACC_TAG(AccGpuSyclIntelTag);
+    CREATE_ACC_TAG(AccOaccTag);
+    CREATE_ACC_TAG(AccOmp5Tag);
+
+    template<typename TAcc>
+    struct AccToTagTrait;
```

Comment:
IMO all traits should be in the namespace trait and not mangled with the word trait.
Else it will not follow the structure we have already in alpaka for other rtraits.

```C++
namespace trait
{
  template<typename TAcc>
    struct AccToTag;

template<typename TTag, typename TDim, typename TIdx>
    struct TagToAcc;
}
```

Reply:
Makes sense. In the first implementation, it didn't made sense but now it is useful. I will change the relic ;-)